cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -178,8 +228,120 @@ public static Double getSizeInMiB(String size) {
      * @param throughput The String representation of the throughput.
      * @return The equivalent number of kibibytes.
      */
-    public static Double getThroughputInKiB(String throughput) {
+    public static String getThroughputInKiB(String throughput) {
         String size = throughput.substring(0, throughput.indexOf("B"));
-        return StorageUtils.convertTo(size, "Ki");
+        return String.valueOf(StorageUtils.convertTo(size, "Ki"));
+    }
+
+    public static void main(String[] args) {
+        System.out.println(getThroughputInKiB("10000KiB/s"));
+        System.out.println(removeKibSuffix("10000KiB/s"));
+
+    }
+
+    private static Double removeKibSuffix(String s) {
+        return Double.valueOf(s.substring(0, s.length() - 5));
+    }
+
+    private void processCapacityEntries(BrokerCapacity bc, Storage s) {
+        String cpu = processCpu();
+        String disk = processDisk(storage, Broker.DEFAULT_BROKER_ID);
+        String inboundNetwork = processInboundNetwork(bc, null);
+        String outboundNetwork = processOutboundNetwork(bc, null);
+
+        // Default broker entry
+        Broker defaultBroker = new Broker(Broker.DEFAULT_BROKER_ID, cpu, disk, inboundNetwork, outboundNetwork);
+        capacityEntries.put(Broker.DEFAULT_BROKER_ID, defaultBroker);
+
+        if (storage instanceof JbodStorage) {
+            // A capacity configuration for a cluster with a JBOD configuration
+            // requires a distinct broker capacity entry for every broker because the
+            // Kafka volume paths are not homogeneous across brokers and include
+            // the broker pod index in their names.
+            for (int id = 0; id < replicas; id++) {
+                disk = processDisk(storage, id);
+                Broker broker = new Broker(id, cpu, disk, inboundNetwork, outboundNetwork);
+                capacityEntries.put(id, broker);
+            }
+        }
```

Comment:
Maybe I am missing something here but ... it seems that if JBOD is used, you are already filling the entries map will ALL the brokers in the cluster. So it seems that even if you don't have any override, you are anyway filling this entries list with all brokers. Is that correct?

Another point is that on line 285, you have the `processDisk` again and inside that method you check again if storage is JBOD and apply the corresponding logic. Isn't there any duplication here?

Reply:
> Maybe I am missing something here but ... it seems that if JBOD is used, you are already filling the entries map will ALL the brokers in the cluster. So it seems that even if you don't have any override, you are anyway filling this entries list with all brokers. Is that correct?

If JBOD is used then we must have an entry for every broker. Otherwise, we only need a broker entry per override. 

> Another point is that on line 285, you have the processDisk again and inside that method you check again if storage is JBOD and apply the corresponding logic. Isn't there any duplication here?

Yes, you are right, this is a duplicate call that is not needed, just removed it now!

Reply:
Wait ... I thought that we could remove the big block of code here not the call on line 285.
So now I have two doubts:

Why, if JBOD is used, we MUST HAVE an entry for every broker? Can you elaborate or giving me a reference with reasons.

If JBOD is not used, this block doesn't run, but now you removed the `processDisk` on line 285, so disk info won't be set anymore?

Reply:
> Why, if JBOD is used, we MUST HAVE an entry for every broker? Can you elaborate or giving me a reference with reasons.

A capacity configuration for a cluster with a JBOD configuration requires a distinct broker capacity entry for every broker because the Kafka volume paths are not homogeneous across brokers and include the broker pod index in their names like `/var/lib/kafka<volume_number>/kafka-log<broker_id>`. For Cruise Control to find the JBOD volumes we must list them by path in the `DISK` sections of the broker entries like this:
```
     {
        "brokerId": "0",
        "capacity": {
          "DISK": {
              "/var/lib/kafka0/kafka-log0": "100000", 
              "/var/lib/kafka1/kafka-log0": "200000"
            },
            ...
        },
        "doc": "Capacity for Broker 0"
      }
        "brokerId": "1",
        "capacity": {
          "DISK": {
              "/var/lib/kafka0/kafka-log1": "100000",
              "/var/lib/kafka1/kafka-log1": "200000"
            },
            ...
        },
        "doc": "Capacity for Broker 1"
      },
 ```

> If JBOD is not used, this block doesn't run, but now you removed the processDisk on line 285, so disk info won't be set anymore?

If JBOD is not used then all brokers will have the same disk capacity configuration as set at the beginning of the method on line 239.  The overriden brokers will have their own broker entries with this disk value and the other brokers will rely on the same disk value as they use the default broker entry configuration